pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -334,55 +232,35 @@ def add_config(self, config: SimConfig) -> None:
             raise ValueError(f"Object {config} is not a valid `SimConfig`")
 
         not_supported = (
-            set(config.noise) - config.supported_noises[self._interaction]
+            set(config.noise)
+            - config.supported_noises[self._hamiltonian._interaction]
         )
         if not_supported:
             raise NotImplementedError(
-                f"Interaction mode '{self._interaction}' does not support "
-                f"simulation of noise types: {', '.join(not_supported)}."
+                f"Interaction mode '{self._hamiltonian._interaction}' does not"
+                " support simulation of noise types: "
+                f"{', '.join(not_supported)}."
             )
-
-        old_noise_set = set(self.config.noise)
-        new_noise_set = old_noise_set.union(config.noise)
-        diff_noise_set = new_noise_set - old_noise_set
-        # Create temporary param_dict to add noise parameters:
-        param_dict: dict[str, Any] = asdict(self._config)
-        # Begin populating with added noise parameters:
-        param_dict["noise"] = tuple(new_noise_set)
-        if "SPAM" in diff_noise_set:
-            param_dict["eta"] = config.eta
-            param_dict["epsilon"] = config.epsilon
-            param_dict["epsilon_prime"] = config.epsilon_prime
-        if "doppler" in diff_noise_set:
-            param_dict["temperature"] = config.temperature
-        if "amplitude" in diff_noise_set:
-            param_dict["laser_waist"] = config.laser_waist
-        if "dephasing" in diff_noise_set:
-            param_dict["dephasing_prob"] = config.dephasing_prob
-        if "depolarizing" in diff_noise_set:
-            param_dict["depolarizing_prob"] = config.depolarizing_prob
-        if "eff_noise" in diff_noise_set:
-            param_dict["eff_noise_opers"] = config.eff_noise_opers
-            param_dict["eff_noise_probs"] = config.eff_noise_probs
-        param_dict["temperature"] *= 1.0e6
-        # update runs:
-        param_dict["runs"] = config.runs
-        param_dict["samples_per_run"] = config.samples_per_run
-
-        # set config with the new parameters:
-        self.set_config(SimConfig(**param_dict))
+        print("transformed", config.to_noise_model())
+        self._hamiltonian.add_config(config.to_noise_model())
 
     def show_config(self, solver_options: bool = False) -> None:
         """Shows current configuration."""
-        print(self._config.__str__(solver_options))
+        print(self.config.__str__(solver_options))
 
     def reset_config(self) -> None:
         """Resets configuration to default."""
-        self.set_config(SimConfig())
+        self._hamiltonian.set_config(NoiseModel())
```

Comment:
Just to be safe, I would rather do 
```suggestion
        self._hamiltonian.set_config(SimConfig().to_noise_model())
```